toolkit/tools/graphanalytics/graphanalytics.go

```diff
@@ -0,0 +1,295 @@
+			}
+
+			// Skip requirements that come from the same srpm.
+			dependencySRPM := filepath.Base(dependency.SrpmPath)
+			if pkgSRPM == dependencySRPM {
+				continue
+			}
+
+			dependencyName := nodeDependencyName(dependency)
+			insertIfMissing(srpmsBlockedBy, pkgSRPM, dependencyName)
+		}
+	}
+
+	printTitle("[DIRECT] SRPMs closest to being ready to build")
+	printReversedMap(srpmsBlockedBy, "unmet dependencies", maxResults)
+}
+
+// printDirectlyClosestToBeingUnblocked will print the packages with the fewest unresolved indrect build requires.
+func printIndirectlyClosestToBeingUnblocked(pkgGraph *pkggraph.PkgGraph, maxResults int) {
+	srpmsBlockedBy := make(map[string][]string)
+
+	for _, node := range pkgGraph.AllNodes() {
+		if node.Type != pkggraph.TypeBuild {
+			continue
+		}
+
+		if node.State == pkggraph.StateUnresolved {
+			continue
+		}
+
+		pkgSRPM := filepath.Base(node.SrpmPath)
+
+		search := traverse.BreadthFirst{}
+		search.Walk(pkgGraph, node, func(n graph.Node, d int) (stopSearch bool) {
+			dependency := n.(*pkggraph.PkgNode)
+
+			// Only consider unresolved build nodes.
+			if dependency.State != pkggraph.StateUnresolved && dependency.State != pkggraph.StateBuild {
+				return
+			}
+
+			// Skip requirements that come from the same srpm.
+			dependencySRPM := filepath.Base(dependency.SrpmPath)
+			if pkgSRPM == dependencySRPM {
+				return
+			}
+
+			dependencyName := nodeDependencyName(dependency)
+			insertIfMissing(srpmsBlockedBy, pkgSRPM, dependencyName)
+
+			return
+		})
+
+	}
+
+	printTitle("[INDIRECT] SRPMs closest to being ready to build")
+	printReversedMap(srpmsBlockedBy, "total unmet dependencies", maxResults)
+}
+
+// nodeDependencyName returns a common dependency name for a node that will be shared across similair Meta/Run/Build nodes for the same package.
+func nodeDependencyName(node *pkggraph.PkgNode) (name string) {
+	// Prefer the SRPM name if possible, otherwise use the unversioned package name
+	srpmName := filepath.Base(node.SrpmPath)
+	if srpmName == "" || srpmName == "<NO_SRPM_PATH>" {
+		name = node.VersionedPkg.Name
+	} else {
+		name = srpmName
+	}
+
+	return
+}
+
+// printTitle prints a formatted title
+func printTitle(title string) {
+	logger.Log.Info("")
+	logger.Log.Info("================================================")
+	logger.Log.Info(title)
+	logger.Log.Info("================================================")
+}
+
+// sortMap returns a sorted slice of a map. If inverse is set, it will return the smallest entries first.
+// It will sort entries by the number of values each key has. For keys with the same number of entries, it will
+// sort alphabetically.
+func sortMap(mapToSort map[string][]string, inverse bool) (pairList []mapPair) {
+	pairList = make([]mapPair, 0, len(mapToSort))
+	for key, value := range mapToSort {
+		pairList = append(pairList, mapPair{key, value})
+	}
+
+	sort.Slice(pairList, func(i, j int) bool {
+		if len(pairList[i].value) == len(pairList[j].value) {
```

Comment:
>len(pairList[i].value) [](start = 5, length = 22)

Suggestion: to make it a bit easier to read (and for a slight perf improvement) we could save the lengths into variables first. Now it takes a bit more focus to figure out what we're comparing in each `if`.